vllm/engine/arg_utils.py

```diff
@@ -1701,7 +1701,12 @@ def _set_default_args_v1(self, usage_context: UsageContext,
 @dataclass
 class AsyncEngineArgs(EngineArgs):
     """Arguments for asynchronous vLLM engine."""
-    disable_log_requests: bool = False
+    # Request logging is disabled by default.  ``--disable-log-requests`` is
+    # kept for backwards compatibility but has no effect.  ``--enable-legacy-
+    # log-requests`` can be used to restore the previous behaviour of logging
+    # each request.
+    disable_log_requests: bool = True
+    enable_legacy_log_requests: bool = False
```

Comment:
Is it necessary to add a new arg? Flipping the default should be enough and then to re-enable it users would use `--no-disable-log-requests`

```suggestion
    disable_log_requests: bool = True
```

Reply:
Please go ahead.

Reply:
https://github.com/vllm-project/vllm/pull/21739